client/components/AvailabilityGrid/AvailabilityGrid.js

```diff
@@ -680,21 +679,78 @@ AvailabilityGrid.defaultProps = {
-  dates: React.PropTypes.arrayOf(React.PropTypes.object).isRequired,
-  heatmap: React.PropTypes.bool,
-  curUser: React.PropTypes.shape({
-    _id: React.PropTypes.string,
-    name: React.PropTypes.string,
-    avatar: React.PropTypes.string,
-  }),
-  availability: React.PropTypes.arrayOf(React.PropTypes.array).isRequired,
-  submitAvail: React.PropTypes.func,
-  closeGrid: React.PropTypes.func,
-  editAvail: React.PropTypes.func,
-  myAvailability: React.PropTypes.arrayOf(React.PropTypes.array),
-  participants: React.PropTypes.arrayOf(React.PropTypes.object),
-  event: React.PropTypes.shape({
-    participants: React.PropTypes.array,
+  // List of dates ranges for event
+  dates: PropTypes.arrayOf(PropTypes.shape({
+    fromDate: PropTypes.instanceOf(Date),
+    toDate: PropTypes.instanceOf(Date),
+  })).isRequired,
+
+  // True if grid is showing heat map
+  heatmap: PropTypes.bool,
+
+  // Current user
+  curUser: PropTypes.shape({
+    _id: PropTypes.string,      // Unique user id
+    name: PropTypes.string,     // User name
+    avatar: PropTypes.string,   // URL to image representing user(?)
+  }).isRequired,
+
+  // List of list of availability times used for heat map
+  availability: PropTypes.arrayOf(PropTypes.arrayOf(
+    PropTypes.arrayOf(PropTypes.string))).isRequired,
+
+  // Function to run when availability for current user is ready to be updated
+  submitAvail: PropTypes.func.isRequired,
+
+  // Function to run when user wishes to cancel availability editing
+  closeGrid: PropTypes.func.isRequired,
+
+  // Function to run to switch from heat map to availability editing
+  editAvail: PropTypes.func.isRequired,
+
+  // Current user's availability array
+  myAvailability: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string)),
+
+  // List of participants in event (dup of contents of event?)
+  participants: PropTypes.arrayOf(PropTypes.shape({
+    userId: PropTypes.shape({
+      id: PropTypes.string,
+      avatar: PropTypes.string,
+      name: PropTypes.string,
+      emails: PropTypes.arrayOf(PropTypes.string),
+    }),
+    _id: PropTypes.string,
+    status: PropTypes.number,
+    emailUpdate: PropTypes.bool,
+    ownerNotified: PropTypes.bool,
+    availability: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string)),
+  })),
+
+  // Event containing list of event participants
+  event: PropTypes.shape({
+    _id: PropTypes.string,
+    name: PropTypes.string,
+    owner: PropTypes.string,
+    active: PropTypes.bool,
+    selectedTimeRange: PropTypes.array,
+    dates: PropTypes.arrayOf(PropTypes.shape({
+      fromDate: PropTypes.string,
+      toDate: PropTypes.string,
+      _id: PropTypes.string,
+    })),
+    participants: PropTypes.arrayOf(PropTypes.shape({
+      userId: PropTypes.shape({
+        id: PropTypes.string,
+        avatar: PropTypes.string,
+        name: PropTypes.string,
+        emails: PropTypes.arrayOf(PropTypes.string),
+      }),
+      _id: PropTypes.string,
+      status: PropTypes.number,
```

Comment:
status you can use PropTypes.oneOf([0.1.2.3])
Because : 
 /**
   *  here whe define the status os the user-guest in
   *  relaction to this Event
   *  0 inactive - when the guest is deleted from the Event
   *  1 invited
   *  2 joined the Event - --default--
   *  3 has added hes time table
   */